res/controllers/Denon-MC7000-scripts.js

```diff
@@ -816,25 +849,78 @@ MC7000.sortLibrary = function(channel, control, value) {
+            midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, vuLevelOutValue);
+            MC7000.prevVuLevel[deckNumber - 1] = vuThreshold;
+        }
+    } else { // if not playing set zero, otherwise 1st LED stays active when < threashold
+        midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, 0x00);
+    }
 };
 
-/* LEDs around Jog wheel */
-MC7000.JogLed = function(value, group) {
+// Spinning Platter LEDs & Slicer Loop PAD LEDs as beat counter
+// pulled together for the calculation to be done only once and then
+// send LED signals to Jog or PAD LEDs when needed.
+MC7000.TrackPositionLEDs = function(value, group) {
     // do nothing before track starts
     if (value === 0) {
         return;
     }
+    // lets define some variables first
     var deckNumber = script.deckFromGroup(group),
-        trackDuration = engine.getValue(group, "duration"),
-        position = value * trackDuration / 60 * MC7000.scratchParams.recordSpeed,
-        // LED ring contains 48 segments with each LED activated by the next even number
-        jogLedOutValue = 48 * 2,
-        activeLED = MC7000.isVinylMode[deckNumber - 1] ? Math.round(position * jogLedOutValue) % jogLedOutValue : value * jogLedOutValue;
+        trackDuration = engine.getValue(group, "duration"), // in seconds
+        beatLength = engine.getValue(group, "file_bpm") / 60, // in Beats Per Seconds
+        cuePosition = engine.getValue(group, "cue_point") / engine.getValue(group, "track_samplerate") / 2, // in seconds
+        playPosition = value * trackDuration, // in seconds
+        jogLEDPosition = playPosition / 60 * MC7000.scratchParams.recordSpeed,
+        jogLEDNumber = 48, // LED ring contains 48 segments each triggered by the next even Midi value
+        // check for Vinyl Mode and decide to spin the Jog LED or show play position
+        activeJogLED = MC7000.isVinylMode[deckNumber - 1] ? Math.round(jogLEDPosition * jogLEDNumber) % jogLEDNumber : Math.round(value * jogLEDNumber),
+        // count the beats (1 to 8) behind the CUE point
+        beatCountLED = (Math.floor((playPosition - cuePosition) * beatLength) % 8); //calculate PAD LED position
+
+    // Todo: check for playposition < (trackduration - warning length) for sending position signals
+
+    // check if a Jog LED has changed and if so then send the signal to the next Jog LED
+    if (MC7000.prevJogLED[deckNumber - 1] !== activeJogLED) {
+        midi.sendShortMsg(0x90 + deckNumber - 1, 0x06, activeJogLED * 2); // only each 2nd midi signal triggers the next LED
+        MC7000.prevJogLED[deckNumber - 1] = activeJogLED;
+    }
+
+    // Todo: else blink the platter LEDs
 
-    midi.sendShortMsg(0x90 + deckNumber - 1, 0x06, activeLED);
+    // check if Slicer mode is active and illuminate PAD LEDs counting with the beat while playing
+    if (MC7000.experimental) {
+        if (MC7000.PADModeSlicer[deckNumber] && engine.getValue(group, "play_indicator")) {
```

Comment:
Its only incidental that the `play_indicator` matches the behavior you want. You probably want to use `beat_active` instead, but be aware of the [recently introduced changes to beat_active](https://github.com/mixxxdj/mixxx/pull/3608) between 2.3 and 2.4

Reply:
> If you point me to some higher level documentation

I have discussed that calculation in [zulip chat](https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/.5Bsolved.5D.20calculation.20for.20relative.20track.20position). please find the pictures and footage as well in there

Reply:
I also tried `beat_active` but it was not reliable. it was missing some beats always. I couldnt find a pattern - seemed to be randomly happening

Reply:
> I have discussed that calculation in zulip chat. please find the pictures and footage as well in there.

Thank you, I'll take a closer look when I find the time.

> I also tried beat_active but it was not reliable. it was missing some beats always. I couldnt find a pattern - seemed to be randomly happening.

Thats strange. Using `play_indicator` is still not a good idea because it was not intended for this usecase. Thus its very likely that a change will break your code in the future. I'll look into an alternative solution.

Reply:
Awesome !!! To further improve that code is much appreciated even though in current shape it is working extremely well.

Reply:
The behavior of the `play_indicator` CO depends on the Cue Mode chosen in mixxx. You could probably cause some weird behavior with the current mapping by using the Denon cue mode and then manually changing the playposition by using the jogwheel. 
I don't have an MC7000 so evaluating whether certain solutions would work is difficult. Couldn't you just omit the `&& engine.getValue(group, "play_indicator")` part? The I don't see why the beatcount should only get updated when the track is playing, it can also update just by using the jogwheel when the deck is paused.

Reply:
I just deleted `&& engine.getValue(group, "play_indicator")` so we could go with the LEDs always counting - also with JogWheel turn.